### PR TITLE
Tuning up & ignore crawled fields feature

### PR DESCRIPTION
**Pull request by mvidalgarcia**
- Update old dependencies
- SSL support (dev)
- Feature: Allow to ignore crawled fields.

### CODE REVIEW

File: cephalopod/webinterface/misc.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 import bcrypt
 from collections import Counter
-from flask import render_template, jsonify, g, request, flash, redirect, url_for
+from flask import current_app, render_template, jsonify, g, request, flash, redirect, url_for
 from flask.ext.login import login_user, logout_user, login_required
```
Comment (ThiefMaster) [inline cephalopod/webinterface/misc.py]: `flask_login` (`flask.ext.*` is deprecated)

File: cephalopod/templates/manage_server.html [hunk 2]
```diff
@@ -126,16 +126,16 @@
                         </div>
                         <div class="panel-body container-fluid">
                             {% if server.crawled_data %}
-                                {% for key, value in server.crawled_data | dictsort %}
-                                    {% if value %}
-                                        {% set aggregation_func = config['CRAWLED_FIELDS_SETTINGS'].get(key, {}).get('aggregation', None) %}
+                                {% for field in fields %}
+                                    {% if server.crawled_data[field] %}
+                                        {% set aggregation_func = config['CRAWLED_FIELDS_SETTINGS'].get(field, {}).get('aggregation', None) %}
```
Comment (ThiefMaster) [inline cephalopod/templates/manage_server.html]: `config.CRAWLED_FIELDS_SETTINGS.get(..)`

File: cephalopod/templates/manage_server.html [hunk 3]
```diff
@@ -126,16 +126,16 @@
                         </div>
                         <div class="panel-body container-fluid">
                             {% if server.crawled_data %}
-                                {% for key, value in server.crawled_data | dictsort %}
-                                    {% if value %}
-                                        {% set aggregation_func = config['CRAWLED_FIELDS_SETTINGS'].get(key, {}).get('aggregation', None) %}
+                                {% for field in fields %}
+                                    {% if server.crawled_data[field] %}
```
Comment (ThiefMaster) [inline cephalopod/templates/manage_server.html]: {% for field in fields if server.crawled_data[field] %}

File: cephalopod/webinterface/misc.py [hunk 4]
```diff
@@ -13,10 +13,19 @@
...
-        extra_fields.update(server.crawled_data or {})
+        extra_fields.update(get_server_fields(server) or {})
+    return sorted(extra_fields)
+
+
+def get_server_fields(server):
+    extra_fields = set()
+    fields_settings = current_app.config['CRAWLED_FIELDS_SETTINGS']
+    if server.crawled_data:
+        data = {k: v for k, v in server.crawled_data.iteritems() if fields_settings.get(k, False) is not None}
```
Comment (ThiefMaster) [inline cephalopod/webinterface/misc.py]: Do None/False have different meanings? If False as no special purpose I'd consider anything falsy as "ignored" like you actually already do in the jinja loop above

File: requirements.txt [hunk 5]
```diff
@@ -9,21 +9,23 @@ Jinja2==2.7.2
...
 pytz==2014.2
 redis==2.9.1
-requests==2.2.1
+requests==2.13.0
 speaklater==1.3
-webassets==0.9
+webassets==0.12.1
 wsgiref==0.1.2
+psycopg2==2.7.1
+pyOpenSSL==16.2.0
```
Comment (ThiefMaster) [inline requirements.txt]: why is this a requirement? isn't it only needed to run the devserver with SSL?

File: cephalopod/webinterface/misc.py (hunk 4 above)
Comment (mvidalgarcia) [inline cephalopod/webinterface/misc.py]: I don't follow. The problem is if there would be just `if fields_settings.get(k)` is going to ignore all the fields that are not specified in the config file and this is not the behavior that Cephalopod had previously.

I think there are two approaches we can follow:
1. Taking into account __all__ the crawled fields no matter if they are specified in the config file or not and ignore only the ones that are specified as `None` in the config. In this case the config file would be useful to customize the aggregation functions, chart type, etc. (The approach that I'm following now and quite similar as it was before except the `None` feature)
2. Taking into account __only__ the fields that are specified in the config file and ignore the rest (so the ignore feature wouldn't make sense here)

Not sure if you were referring to this.

File: requirements.txt (hunk 5 above)
Comment (mvidalgarcia) [inline requirements.txt]: Yes, but then should it be installed separately? I can't run [this](https://github.com/mvidalgarcia/cephalopod/blob/d4e6aebcd310b7295701677bee8386b2da8addfe/cephalopod/manage.py#L18) without pyOpenSSL

File: requirements.dev.txt [hunk 6]
```diff
@@ -0,0 +1 @@
+pyOpenSSL==16.2.0
```
Comment (ThiefMaster) [inline requirements.dev.txt]: I wouldn't require a specific version here. It's dev requirements, having the latest one is almost always a safe  choice.

**Pull request merged**